Reflection_Engine/Compute/RunExtentionMethod.cs

```diff
@@ -46,13 +46,12 @@ public static object RunExtentionMethod(object target, string methodName)
                 return m_PreviousInvokedMethods[key].Invoke(null, new object[] { target });
 
             // Otherwise, search for the method and call it if found
-            foreach (MethodInfo method in type.ExtentionMethods(methodName))
+            MethodInfo method = type.ExtentionMethods(methodName).Where(x => x.GetParameters().Length == 1).SortExtensionMethods(type).FirstOrDefault();
+
+            if (method != null)
             {
-                if (method.GetParameters().Length == 1)
-                {
-                    m_PreviousInvokedMethods[key] = method;
-                    return method.Invoke(null, new object[] { target });
-                }
+                m_PreviousInvokedMethods[key] = method;
+                return method.Invoke(null, new object[] { target });
             }
 
             // Return null if nothing found
```

Comment:
Is `result` initialised as `null` and never assigned for a reason or is it just a leftover? 
It happens in the next overload as well.

Reply:
Just a leftover, no real reason.

Could change if you think it is important?